route/build/src/lake_route.f90

```diff
@@ -0,0 +1,137 @@
+  
+
+  ! perform lake routing based on a fixed storage discharge relationship Q=kS
+  ! no runoff input is added to the lake; the input are only precipitation and evaporation to the lake
+
+  print*, '------lake-simulation-------- '
+  print*, 'node id that is lake ...... = ', NETOPO_in(segIndex)%REACHID ! to check the reach id of lake
+  print*, 'lake param RATECVA ........ = ', RPARAM_in(segIndex)%RATECVA
+  print*, 'lake param RATECVB ........ = ', RPARAM_in(segIndex)%RATECVB
+  print*, 'volume before simulation m3 = ', RCHFLX_out(iens,segIndex)%REACH_VOL(0)
+  print*, 'upstream streamflow m3/s .. = ', RCHFLX_out(iens,segIndex)%REACH_Q_IRF
+  print*, 'upstream precipitation m3/s = ', RCHFLX_out(iens,segIndex)%basinprecip
+  print*, 'upstream evaporation m3/s . = ', RCHFLX_out(iens,segIndex)%basinevapo
+  
+  RCHFLX_out(iens,segIndex)%REACH_VOL(1) = RCHFLX_out(iens,segIndex)%REACH_VOL(0) ! updating storage for current time
+  RCHFLX_out(iens,segIndex)%REACH_VOL(1) = RCHFLX_out(iens,segIndex)%REACH_VOL(1) + q_upstream * dt  ! input upstream discharge from m3/s to m3
+  RCHFLX_out(iens,segIndex)%REACH_VOL(1) = RCHFLX_out(iens,segIndex)%REACH_VOL(1) + RCHFLX_out(iens,segIndex)%basinprecip * dt ! input lake precipitation
+  RCHFLX_out(iens,segIndex)%REACH_VOL(1) = RCHFLX_out(iens,segIndex)%REACH_VOL(1) - RCHFLX_out(iens,segIndex)%basinevapo * dt ! output lake evaporaiton
+  if (RCHFLX_out(iens,segIndex)%REACH_VOL(1) .LT. 0) then; ! set the lake volume as 0 if it goes negative actual evaporation is not calculated here in case low storage mean low evaporaiton...
+    RCHFLX_out(iens,segIndex)%REACH_VOL(1)=0
+  endif
+  ! A has the dimnesion of 1/s; the flux leaving per second
+  RCHFLX_out(iens,segIndex)%REACH_Q_IRF = RPARAM_in(segIndex)%RATECVA * (RCHFLX_out(iens,segIndex)%REACH_VOL(1) ** RPARAM_in(segIndex)%RATECVB)! simplified level pool liner reservoir Q=AS^B
+  RCHFLX_out(iens,segIndex)%REACH_Q_IRF = (min(RCHFLX_out(iens,segIndex)%REACH_Q_IRF * dt, RCHFLX_out(iens,segIndex)%REACH_VOL(1)) )/dt! in case is the output volume is more than lake volume
+  RCHFLX_out(iens,segIndex)%REACH_VOL(1) = RCHFLX_out(iens,segIndex)%REACH_VOL(1) - RCHFLX_out(iens,segIndex)%REACH_Q_IRF * dt ! updating the storage
+  if (RCHFLX_out(iens,segIndex)%REACH_VOL(1) .LT. 0) then; ! set the lake volume as 0 if it goes negative
+    RCHFLX_out(iens,segIndex)%REACH_VOL(1)=0
+  endif
+
+
+  print*, 'lake simulated output m3/s .= ', RCHFLX_out(iens,segIndex)%REACH_Q_IRF
+  print*, 'volume after simulation m3 .= ', RCHFLX_out(iens,segIndex)%REACH_VOL(1)
+
+  ! calculate water balance (in this water balance we dont have the actual evaporation)
+  WB = q_upstream * dt + RCHFLX_out(iens,segIndex)%basinprecip * dt - RCHFLX_out(iens,segIndex)%REACH_Q_IRF * dt &
+  - RCHFLX_out(iens,segIndex)%basinevapo * dt - (RCHFLX_out(iens,segIndex)%REACH_VOL(1) - RCHFLX_out(iens,segIndex)%REACH_VOL(0))
+  print*, 'water balance error ....... = ', WB
```

Comment:
For this one,   print out WB when it is above threshold as warning or even terminate  ?

Reply:
ok, here I can do a check with a preexisting values of minimum water balance... I would say for a lake 1 m3 is an acceptable value for each time step. the resulting WB are in order of 10e-9 (which is larger than the 10e-15 for min flow for example). for now I will try to set a value in the Global file for 10e-4 for example...